File: 20180709/ToeplitzMatrix.py
class Solution(object):
    def isToeplitzMatrix(self, matrix):
        """
        :type matrix: List[List[int]]
        :rtype: bool
        """
        M = len(matrix)
        if M < 1:
            return True
        N = len(matrix[0])
        for i in range(M-1):
            for j in range(N - 1):
                if matrix[i][j] != matrix[i+1][j+1]:
                    return False
        return True

        

File: 20180709/ToeplitzMatrix.java
class Solution {
    public boolean isToeplitzMatrix(int[][] matrix) {
        int M = matrix.length;
        if(M < 1){
            return true;
        }
        int N = matrix[0].length;
        for(int i = 0; i < M - 1; i++){
            for(int j = 0; j < N - 1; j++){
                if (matrix[i][j] != matrix[i+1][j+1]){
                    return false;
                }
            }
        }
        return true;
    }
}
